src/Aspire.Dashboard/ResourceService/Partials.cs

```diff
@@ -24,7 +24,15 @@ public ResourceViewModel ToViewModel()
                 DisplayName = ValidateNotNull(DisplayName),
                 Uid = ValidateNotNull(Uid),
                 CreationTimeStamp = ValidateNotNull(CreatedAt).ToDateTime(),
-                Properties = Properties.ToFrozenDictionary(property => ValidateNotNull(property.Name), property => ValidateNotNull(property.Value), StringComparers.ResourcePropertyName),
+                Properties = Properties.ToFrozenDictionary(
+                    comparer: StringComparers.ResourcePropertyName,
+                    keySelector: property => ValidateNotNull(property.Name),
+                    elementSelector: property => new ResourcePropertyViewModel(
+                        name: ValidateNotNull(property.Name),
+                        value: ValidateNotNull(property.Value),
+                        isValueSensitive: property.IsSensitive,
+                        knownProperty: null, // TODO can we do better here? yes, if the server sends known properties
```

Comment:
In future I hope to set this and the priority value directly here, simplifying the code a little by removing the need for `PopulateMissingKnownProperties`.

Reply:
This was done in this PR.